src/Orleans/LogConsistency/ILogConsistencyProtocolServices.cs

```diff
@@ -0,0 +1,145 @@
+﻿
+using System;
+using System.Net;
+using System.Runtime.Serialization;
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Orleans.Providers;
+using Orleans.Runtime;
+using Orleans.MultiCluster;
+using Orleans.GrainDirectory;
+
+namespace Orleans.LogConsistency
+{
+    /// <summary>
+    /// Functionality for use by log view adaptors that use custom consistency or replication protocols.
+    /// Abstracts communication between replicas of the log-consistent grain in different clusters.
+    /// </summary>
+    public interface ILogConsistencyProtocolServices
+    {
+        /// <summary>
+        /// Send a message to a remote cluster.
+        /// </summary>
+        /// <param name="payload">the message</param>
+        /// <param name="clusterId">the destination cluster id</param>
+        /// <returns></returns>
+        Task<ILogConsistencyProtocolMessage> SendMessage(ILogConsistencyProtocolMessage payload, string clusterId);
+
+
+        /// <summary>
+        /// The untyped reference for this grain.
+        /// </summary>
+        GrainReference GrainReference { get;  }
+
+        
+        /// <summary>
+        /// The multicluster registration strategy for this grain.
+        /// </summary>
+        IMultiClusterRegistrationStrategy RegistrationStrategy { get; }
+
+
+        /// <summary>
+        /// Whether this cluster is running in a multi-cluster network.
+        /// </summary>
+        /// <returns></returns>
+        bool MultiClusterEnabled { get; }
+
+        // If there is no multi-cluster network,
+        // we use a default multi-cluster configuration containing just one cluster (this one), named "I"
+        // this is more convenient and consistent than returning null or invalid,
+        // since it means the log-consistency providers do the right thing (run like a single-cluster configuration)
+
+        /// <summary>
+        /// The id of this cluster. Returns "I" if no multi-cluster network is present.
+        /// </summary>
+        /// <returns></returns>
+        string MyClusterId { get; }
+
```

Comment:
DeploymentId is indeed always non-null. I completely agree that ClusterId is a better name. I did not mean that we should consolidate the two in this PR, we should discuss in its own issue. 
Perhaps since ClusterId is a better name, we can get rid of DeploymentId in the much bigger breaking change release (2.0), since config will probably be changed too.